src/conn.c

```diff
@@ -2153,7 +2158,7 @@ _processOpError(natsConnection *nc, natsStatus s, bool initialConnect)
 
     // Do reconnect only if allowed and we were actually connected
     // or if we are retrying on initial failed connect.
-    if (initialConnect || (nc->opts->allowReconnect && (nc->status == NATS_CONN_STATUS_CONNECTED)))
+    if (forceReconnect || (nc->opts->allowReconnect && (nc->status == NATS_CONN_STATUS_CONNECTED)))
```

Comment:
Again, in my mind, we should not need that, we would fall in the nc->opts->allowReconnect and connected case if we were to change `natsConnection_Reconnect()` to check that we are allowed to do this and if not fail the call.